CDTDatastoreReplicationAcceptanceTests/Attachments.m

```diff
@@ -123,131 +110,106 @@ - (void)deleteAttachmentNamed:(NSString*)attachmentName
 #pragma mark Tests
 -(void)testSavedHttpAttachmentWithRemote
 {
-    //
-    // Set up remote database
-    //
-    
-    NSString *remoteDbName = [NSString stringWithFormat:@"%@-test-database-%@",
-                              self.remoteDbPrefix,
-                              [CloudantReplicationBase generateRandomString:5]];
-    NSURL *remoteDbURL = [self.remoteRootURL URLByAppendingPathComponent:remoteDbName];
-    
-    [self createRemoteDatabase:remoteDbName
-                   instanceURL:self.remoteRootURL];
-    
-    //
-    // Add attachments to a document in the local store
-    //
-    
-    // Contains {attachmentName: attachmentContent} for later checking
-    NSMutableDictionary *originalAttachments = [NSMutableDictionary dictionary];
-    
-    NSError *error;
-    
-    NSString *docId = @"document1";
-    CDTDocumentRevision *rev = [CDTDocumentRevision revisionWithDocId:docId];
-    rev.body = [@{ @"hello" : @12 } mutableCopy];
+    if([self.iamApiKey length] != 0) {
+        NSLog(@"IAM API key enabled, skipping test");
```

Comment:
I spoke with @tomblench about this testing failing.
His response on Slack:
...`[CDTSavedHTTPAttachment dataFromAttachmentContent]` is trying to download 
the data outside `CDTURLSession` and fails because it doesn’t have the bearer token...
...but i would like to get rid of the whole class because it’s deprecated...
...it was originally intended to support something on worklight/mobilefirst: https://github.com/cloudant/CDTDatastore/blob/33c27b00e577a97420f53807d77218a8f6cc6f04/CDTDatastore/Attachments/CDTAttachment.h#L122-L125

@tomblench should I go ahead and create a PR to remove this class and test?

Reply:
Yes - we might have to check when it was deprecated. If it was deprecated under 0.x then fine, otherwise it will mean a version bump to 2.x (which I don't think is a problem)

Reply:
https://github.com/cloudant/CDTDatastore/pull/236 - the deprecated messages were added in 1.1.0 (2016-06-17).

Reply:
@tomblench @ricellis PR #391 for removal of deprecated `CDTSavedHTTPAttachment `class.